cosmosdb/cosmosdb_endpoint.bal

```diff
@@ -0,0 +1,725 @@
+                           returns ResourceResponse|error {
+    string resourceLink = DB_PATH + SLASH + databaseId + SLASH + COLL_PATH + SLASH + collectionId;
+    string requestPath = SLASH + resourceLink;
+    http:Request request = new;
+    var requestHeaders = costructRequestHeaders(request, "delete", COLL_PATH, resourceLink, self.masterKey,
+        JSON_CONTENT_TYPE, requestOptions = requestOptions);
+    if (requestHeaders is error) {
+        error err = error(COSMOS_DB_ERROR_CODE,
+        { message: "Error occurred while constructing request headers" });
+        return err;
+    } else {
+        var httpResponse = self.documentClient->delete(requestPath, request);
+        if (httpResponse is http:Response) {
+            int statusCode = httpResponse.statusCode;
+            var jsonResponse = httpResponse.getJsonPayload();
+            if (jsonResponse is json) {
+                if (statusCode == http:NO_CONTENT_204) {
+                    ResourceResponse resourceResponse = extractResponseHeaders(httpResponse);
+                    return resourceResponse;
+                } else {
+                    return setResponseError(jsonResponse);
+                }
+            } else {
+                error err = error(COSMOS_DB_ERROR_CODE,
+                { message: "Error occurred while accessing the JSON payload of the response" });
+                return err;
+            }
+        } else {
+            error err = error(COSMOS_DB_ERROR_CODE, { message: "Error occurred while invoking the REST API" });
+            return err;
+        }
+    }
+}
+
+remote function Client.createDocument(string databaseId, string collectionId, string documentId,
+                                      DocumentCreateOptions? documentCreateOptions = (), json content)
+                           returns ResourceResponse|error {
+    string resourceLink = DB_PATH + SLASH + databaseId + SLASH + COLL_PATH + SLASH + collectionId;
+    string requestPath = SLASH + resourceLink + SLASH + DOC_PATH;
+    http:Request request = new;
+    content["id"] = documentId;
+    request.setJsonPayload(content);
+    var requestHeaders = costructRequestHeaders(request, "post", DOC_PATH, resourceLink, self.masterKey,
+        JSON_CONTENT_TYPE, requestOptions = documentCreateOptions.requestOptions);
+    if (requestHeaders is error) {
+        error err = error(COSMOS_DB_ERROR_CODE,
+        { message: "Error occurred while constructing request headers" });
+        return err;
+    } else {
+        //Set collection specific headers.
+        if (documentCreateOptions != ()) {
+            if (documentCreateOptions.isDocumentUpsert != ()) {
+                request.setHeader("x-ms-documentdb-is-upsert", <string>documentCreateOptions.isDocumentUpsert);
+            }
+            if (documentCreateOptions.indexingDirective != ()) {
+                request.setHeader("x-ms-indexing-directive", <string>documentCreateOptions.indexingDirective);
```

Comment:
Shall we follow the same convention for all the headers? Some are defined as constants and others not. Check all the places.